compiler/rustc_hir_analysis/src/check/errs.rs

```diff
@@ -56,25 +126,32 @@ fn handle_static_mut_ref(
     e2024: bool,
     mutable: Mutability,
     hir_id: hir::HirId,
+    suggest_addr_of: bool,
 ) {
+    let shared = if mutable == Mutability::Mut { "mutable" } else { "shared" };
     if e2024 {
-        let (sugg, shared) = if mutable == Mutability::Mut {
-            (errors::StaticMutRefSugg::Mut { span, var }, "mutable")
-        } else {
-            (errors::StaticMutRefSugg::Shared { span, var }, "shared")
-        };
+        let sugg = suggest_addr_of.then(|| {
+            if matches!(mutable, Mutability::Mut) {
+                errors::StaticMutRefSugg::Mut { span, var }
+            } else {
+                errors::StaticMutRefSugg::Shared { span, var }
+            }
+        });
         tcx.dcx().emit_err(errors::StaticMutRef { span, sugg, shared });
-    } else {
-        let (sugg, shared) = if mutable == Mutability::Mut {
-            (errors::RefOfMutStaticSugg::Mut { span, var }, "mutable")
-        } else {
-            (errors::RefOfMutStaticSugg::Shared { span, var }, "shared")
-        };
-        tcx.emit_node_span_lint(
-            STATIC_MUT_REFS,
-            hir_id,
-            span,
-            errors::RefOfMutStatic { span, sugg, shared },
-        );
+        return;
     }
+
+    let sugg = suggest_addr_of.then(|| {
+        if matches!(mutable, Mutability::Mut) {
+            errors::RefOfMutStaticSugg::Mut { span, var }
+        } else {
+            errors::RefOfMutStaticSugg::Shared { span, var }
```

Comment:
Is there any difference between `errors::RefOfMutStaticSugg` and `errors::StaticMutRefSugg`?

Reply:
Nice catch! You’re right; there is no difference between `errors::RefOfMutStaticSugg` and `errors::StaticMutRefSugg`. They were duplicates, and I have removed the redundancy. Thank you for pointing it out!